runtime/src/main/scala/org/scalaexercises/exercises/Evaluator.scala

```diff
@@ -97,4 +99,15 @@ class Evaluator(timeout: Duration = 20.seconds) {
       }
     }
   }
+
+  def apply[T](pre: String, code: String): EvalResult[T] = {
+    val allCode = s"""
+                     |$pre
+                     |$code
+                   """.stripMargin
+    val fut = Future({ eval(allCode) })
+    Try(
+      Await.result(fut, timeout)
+    ).getOrElse(EvalResult.Timeout())
```

Comment:
With this construction, I think we might be loosing any failure information from the execution of `eval(...)` and replacing it by a timeout. Don't know the codebase to tell if that should be the intention.